lib/experimental/new_embed.dart

```diff
@@ -614,6 +615,109 @@ class FlashBox {
   void hide() {
     _element.setAttr('hidden');
   }
+
+  void show() {
+    _element.clearAttr('hidden');
+  }
+}
+
+class AnalysisResultsElement extends DElement {
+  static const String _noIssuesMsg = 'no issues';
+  static const String _hideMsg = 'hide';
+  static const String _showMsg = 'show';
+
+  DElement message;
+  DElement toggle;
+  bool _hidden = true;
+
+  final StreamController<AnalysisIssue> _onClickSink =
+      StreamController.broadcast();
+  Stream<AnalysisIssue> get onIssueClick => _onClickSink.stream;
+
+  AnalysisResultsElement(Element element, this.message, this.toggle)
+      : super(element) {
+    hide();
+    message.text = _noIssuesMsg;
+    toggle.text = _hideMsg;
+    toggle.onClick.listen((_) {
+      if (_hidden) {
+        show();
+      } else {
+        hide();
+      }
+    });
+  }
+
+  void display(List<AnalysisIssue> issues) {
+    if (issues.isEmpty) {
+      message.text = _noIssuesMsg;
+      hideToggle();
+      return;
+    }
+
+    showToggle();
+    message.text = '${issues.length} issues';
+
+    clearChildren();
+    for (var elem in _createElements(issues)) {
+      add(elem);
+    }
+  }
+
+  Iterable<Element> _createElements(Iterable<AnalysisIssue> issues) {
+    return issues.map(_issueElement);
+  }
+
+  Element _issueElement(AnalysisIssue issue) {
+    var message = issue.message;
+    if (issue.message.endsWith('.')) {
+      message = message.substring(0, message.length - 1);
+    }
+    var elem = DivElement()..classes.add('issue');
+
+    if (issue.kind == 'info') {
```

Comment:
Consider creating a const map of issue.kind strings to CSS class names, and using it to reduce this to one statement:

```dart
elem.children.add(SpanElement()
  ..text = issue.kind
  ..classesAddAll(['issueLabel', cssClassForKind[issue.kind]]));
```